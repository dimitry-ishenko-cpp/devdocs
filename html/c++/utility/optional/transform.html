    <h1 id="firstHeading" class="firstHeading">std::optional&lt;T&gt;::transform</h1>            <table class="t-dcl-begin"> <tr class="t-dcl t-since-cxx23"> <td> <pre data-language="cpp">template&lt; class F &gt;
constexpr auto transform( F&amp;&amp; f ) &amp;;</pre>
</td> <td> (1) </td> <td> <span class="t-mark-rev t-since-cxx23">(since C++23)</span> </td> </tr> <tr class="t-dcl t-since-cxx23"> <td> <pre data-language="cpp">template&lt; class F &gt;
constexpr auto transform( F&amp;&amp; f ) const&amp;;</pre>
</td> <td> (2) </td> <td> <span class="t-mark-rev t-since-cxx23">(since C++23)</span> </td> </tr> <tr class="t-dcl t-since-cxx23"> <td> <pre data-language="cpp">template&lt; class F &gt;
constexpr auto transform( F&amp;&amp; f ) &amp;&amp;;</pre>
</td> <td> (3) </td> <td> <span class="t-mark-rev t-since-cxx23">(since C++23)</span> </td> </tr> <tr class="t-dcl t-since-cxx23"> <td> <pre data-language="cpp">template&lt; class F &gt;
constexpr auto transform( F&amp;&amp; f ) const&amp;&amp;;</pre>
</td> <td> (4) </td> <td> <span class="t-mark-rev t-since-cxx23">(since C++23)</span> </td> </tr>  </table> <p>If <code>*this</code> contains a value, invokes <code>f</code> with the contained value as an argument, and returns an <code>std::optional</code> that contains the result of that invocation; otherwise, returns an empty <code>std::optional</code>.</p>
<p>The type of contained value in the result (denoted by <code>U</code> below) must be a non-array object type, and must not be <code><a href="../in_place.html" title="cpp/utility/in place">std::in_place_t</a></code> or <code><a href="nullopt_t.html" title="cpp/utility/optional/nullopt t">std::nullopt_t</a></code>). Otherwise, the program is ill-formed.</p>
<div class="t-li1">
<span class="t-li">1)</span> Let <code>U</code> be <code><a href="http://en.cppreference.com/w/cpp/types/remove_cv"><span class="kw600">std::<span class="me2">remove_cv_t</span></span></a><span class="sy1">&lt;</span><a href="http://en.cppreference.com/w/cpp/types/result_of"><span class="kw3219">std::<span class="me2">invoke_result_t</span></span></a><span class="sy1">&lt;</span>F, T<span class="sy3">&amp;</span><span class="sy1">&gt;&gt;</span></code>. If <code>*this</code> contains a value, returns a <code><a href="http://en.cppreference.com/w/cpp/utility/optional"><span class="kw1077">std::<span class="me2">optional</span></span></a><span class="sy1">&lt;</span>U<span class="sy1">&gt;</span></code> whose contained value is <a href="../../language/direct_initialization.html" title="cpp/language/direct initialization">direct-initialized</a> from <code><a href="http://en.cppreference.com/w/cpp/utility/functional/invoke"><span class="kw1063">std::<span class="me2">invoke</span></span></a><span class="br0">(</span><a href="http://en.cppreference.com/w/cpp/utility/forward"><span class="kw1120">std::<span class="me2">forward</span></span></a><span class="sy1">&lt;</span>F<span class="sy1">&gt;</span><span class="br0">(</span>f<span class="br0">)</span>, <span class="sy2">**</span>this<span class="br0">)</span></code> (unlike <a href="and_then.html" title="cpp/utility/optional/and then"><code>and_then()</code></a>, which must return an <code><a href="../optional.html" title="cpp/utility/optional">std::optional</a></code> directly). Otherwise, returns an empty <code><a href="http://en.cppreference.com/w/cpp/utility/optional"><span class="kw1077">std::<span class="me2">optional</span></span></a><span class="sy1">&lt;</span>U<span class="sy1">&gt;</span></code>.<br>The program is ill-formed if the variable definition <code>U x<span class="br0">(</span><a href="http://en.cppreference.com/w/cpp/utility/functional/invoke"><span class="kw1063">std::<span class="me2">invoke</span></span></a><span class="br0">(</span><a href="http://en.cppreference.com/w/cpp/utility/forward"><span class="kw1120">std::<span class="me2">forward</span></span></a><span class="sy1">&lt;</span>F<span class="sy1">&gt;</span><span class="br0">(</span>f<span class="br0">)</span>, <span class="sy2">**</span>this<span class="br0">)</span><span class="br0">)</span><span class="sy4">;</span></code> is ill-formed.</div> <div class="t-li1">
<span class="t-li">2)</span> Same as <span class="t-v">(1)</span>, except that <code>U</code> is <code><a href="http://en.cppreference.com/w/cpp/types/remove_cv"><span class="kw600">std::<span class="me2">remove_cv_t</span></span></a><span class="sy1">&lt;</span><a href="http://en.cppreference.com/w/cpp/types/result_of"><span class="kw3219">std::<span class="me2">invoke_result_t</span></span></a><span class="sy1">&lt;</span>F, <span class="kw4">const</span> T<span class="sy3">&amp;</span><span class="sy1">&gt;&gt;</span></code>.</div> <div class="t-li1">
<span class="t-li">3)</span> Let <code>U</code> be <code><a href="http://en.cppreference.com/w/cpp/types/remove_cv"><span class="kw600">std::<span class="me2">remove_cv_t</span></span></a><span class="sy1">&lt;</span><a href="http://en.cppreference.com/w/cpp/types/result_of"><span class="kw3219">std::<span class="me2">invoke_result_t</span></span></a><span class="sy1">&lt;</span>F, T<span class="sy1">&gt;&gt;</span></code>. If <code>*this</code> contains a value, returns a <code><a href="http://en.cppreference.com/w/cpp/utility/optional"><span class="kw1077">std::<span class="me2">optional</span></span></a><span class="sy1">&lt;</span>U<span class="sy1">&gt;</span></code> whose contained value is direct-initialized from <code><a href="http://en.cppreference.com/w/cpp/utility/functional/invoke"><span class="kw1063">std::<span class="me2">invoke</span></span></a><span class="br0">(</span><a href="http://en.cppreference.com/w/cpp/utility/forward"><span class="kw1120">std::<span class="me2">forward</span></span></a><span class="sy1">&lt;</span>F<span class="sy1">&gt;</span><span class="br0">(</span>f<span class="br0">)</span>, std<span class="sy4">::</span><span class="me2">move</span><span class="br0">(</span><span class="sy2">**</span>this<span class="br0">)</span><span class="br0">)</span></code>. Otherwise, returns an empty <code><a href="http://en.cppreference.com/w/cpp/utility/optional"><span class="kw1077">std::<span class="me2">optional</span></span></a><span class="sy1">&lt;</span>U<span class="sy1">&gt;</span></code>.<br>The program is ill-formed if the variable definition <code>U x<span class="br0">(</span><a href="http://en.cppreference.com/w/cpp/utility/functional/invoke"><span class="kw1063">std::<span class="me2">invoke</span></span></a><span class="br0">(</span><a href="http://en.cppreference.com/w/cpp/utility/forward"><span class="kw1120">std::<span class="me2">forward</span></span></a><span class="sy1">&lt;</span>F<span class="sy1">&gt;</span><span class="br0">(</span>f<span class="br0">)</span>, std<span class="sy4">::</span><span class="me2">move</span><span class="br0">(</span><span class="sy2">**</span>this<span class="br0">)</span><span class="br0">)</span><span class="br0">)</span><span class="sy4">;</span></code> is ill-formed.</div> <div class="t-li1">
<span class="t-li">4)</span> Same as <span class="t-v">(3)</span>, except that <code>U</code> is <code><a href="http://en.cppreference.com/w/cpp/types/remove_cv"><span class="kw600">std::<span class="me2">remove_cv_t</span></span></a><span class="sy1">&lt;</span><a href="http://en.cppreference.com/w/cpp/types/result_of"><span class="kw3219">std::<span class="me2">invoke_result_t</span></span></a><span class="sy1">&lt;</span>F, <span class="kw4">const</span> T<span class="sy1">&gt;&gt;</span></code>.</div>  <h3 id="Parameters"> Parameters</h3> <table class="t-par-begin"> <tr class="t-par"> <td> f </td> <td> - </td> <td> a suitable function or <a href="../../named_req/callable.html" title="cpp/named req/Callable">Callable</a> object whose call signature returns a non-reference type </td>
</tr>
</table> <h3 id="Return_value"> Return value</h3> <p>An <code><a href="../optional.html" title="cpp/utility/optional">std::optional</a></code> containing the result of <code>f</code> or an empty <code><a href="../optional.html" title="cpp/utility/optional">std::optional</a></code>, as described above.</p>
<h3 id="Notes"> Notes</h3> <p>Because <code>transform</code> directly constructs a <code>U</code> object at the right location, rather than passing it to a constructor, <code><a href="http://en.cppreference.com/w/cpp/types/is_move_constructible"><span class="kw546">std::<span class="me2">is_move_constructible_v</span></span></a><span class="sy1">&lt;</span>U<span class="sy1">&gt;</span></code> can be <code>false</code>.</p>
<p>As the callable <code>f</code> can't return a reference type, it cannot be a <a href="../../language/pointer.html#Pointers_to_data_members" title="cpp/language/pointer">pointer to data member</a>.</p>
<p>Some languages call this operation <a href="https://en.wikipedia.org/wiki/Map_(higher-order_function)" class="extiw" title="enwiki:Map (higher-order function)"><i>map</i></a>.</p>
<table class="wikitable ftm-begin ftm-has-value"> <tr> <th>
<a href="../feature_test.html" title="cpp/utility/feature test">Feature-test</a> macro</th> <th>Value</th> <th><span title="Standard in which the feature is introduced">Std</span></th> <th>Feature </th>
</tr> <tr> <td><a href="../../feature_test.html#cpp_lib_optional" title="cpp/feature test"><code>__cpp_lib_optional</code></a></td> <td><span class="nu0">202110L</span></td> <td><span class="t-mark">(C++23)</span></td> <td>
<a href="../optional.html#Monadic_operations" title="cpp/utility/optional">Monadic operations</a> in <code><a href="../optional.html" title="cpp/utility/optional">std::optional</a></code> </td>
</tr>
</table> <h3 id="Example"> Example</h3> <div class="t-example"> <div class="cpp source-cpp"><pre data-language="cpp">#include &lt;iostream&gt;
#include &lt;optional&gt;
 
struct A { /* ... */ };
struct B { /* ... */ };
struct C { /* ... */ };
struct D { /* ... */ };
 
auto A_to_B(A) -&gt; B { /* ... */ std::cout &lt;&lt; "A =&gt; B \n"; return {}; }
auto B_to_C(B) -&gt; C { /* ... */ std::cout &lt;&lt; "B =&gt; C \n"; return {}; }
auto C_to_D(C) -&gt; D { /* ... */ std::cout &lt;&lt; "C =&gt; D \n"; return {}; }
 
void try_transform_A_to_D(std::optional&lt;A&gt; o_A)
{
    std::cout &lt;&lt; (o_A ? "o_A has a value\n" : "o_A is empty\n");
 
    std::optional&lt;D&gt; o_D = o_A.transform(A_to_B)
                              .transform(B_to_C)
                              .transform(C_to_D);
 
    std::cout &lt;&lt; (o_D ? "o_D has a value\n\n" : "o_D is empty\n\n");
};
 
int main()
{
    try_transform_A_to_D( A{} );
    try_transform_A_to_D( {} );
}</pre></div> <p>Output:</p>
<div class="text source-text"><pre data-language="c">o_A has a value
A =&gt; B
B =&gt; C
C =&gt; D
o_D has a value
 
o_A is empty
o_D is empty</pre></div> </div> <h3 id="See_also"> See also</h3> <table class="t-dsc-begin"> <tr class="t-dsc"> <td> <div><a href="value_or.html" title="cpp/utility/optional/value or"> <span class="t-lines"><span>value_or</span></span></a></div> </td> <td> returns the contained value if available, another value otherwise <br> <span class="t-mark">(public member function)</span>  </td>
</tr> <tr class="t-dsc"> <td> <div><a href="and_then.html" title="cpp/utility/optional/and then"> <span class="t-lines"><span>and_then</span></span></a></div>
<div><span class="t-lines"><span><span class="t-mark-rev t-since-cxx23">(C++23)</span></span></span></div> </td> <td> returns the result of the given function on the contained value if it exists, or an empty <code>optional</code> otherwise <br> <span class="t-mark">(public member function)</span>  </td>
</tr> <tr class="t-dsc"> <td> <div><a href="or_else.html" title="cpp/utility/optional/or else"> <span class="t-lines"><span>or_else</span></span></a></div>
<div><span class="t-lines"><span><span class="t-mark-rev t-since-cxx23">(C++23)</span></span></span></div> </td> <td> returns the <code>optional</code> itself if it contains a value, or the result of the given function otherwise <br> <span class="t-mark">(public member function)</span>  </td>
</tr> </table>           <div class="_attribution">
  <p class="_attribution-p">
    &copy; cppreference.com<br>Licensed under the Creative Commons Attribution-ShareAlike Unported License v3.0.<br>
    <a href="https://en.cppreference.com/w/cpp/utility/optional/transform" class="_attribution-link">https://en.cppreference.com/w/cpp/utility/optional/transform</a>
  </p>
</div>
