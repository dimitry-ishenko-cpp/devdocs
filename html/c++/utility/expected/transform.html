    <h1 id="firstHeading" class="firstHeading">std::expected&lt;T,E&gt;::transform</h1>            <table class="t-dcl-begin"> <tr class="t-dcl t-since-cxx23"> <td> <pre data-language="cpp">template&lt; class F &gt;
constexpr auto transform( F&amp;&amp; f ) &amp;;</pre>
</td> <td> (1) </td> <td> <span class="t-mark-rev t-since-cxx23">(since C++23)</span> </td> </tr> <tr class="t-dcl t-since-cxx23"> <td> <pre data-language="cpp">template&lt; class F &gt;
constexpr auto transform( F&amp;&amp; f ) const&amp;;</pre>
</td> <td> (2) </td> <td> <span class="t-mark-rev t-since-cxx23">(since C++23)</span> </td> </tr> <tr class="t-dcl t-since-cxx23"> <td> <pre data-language="cpp">template&lt; class F &gt;
constexpr auto transform( F&amp;&amp; f ) &amp;&amp;;</pre>
</td> <td> (3) </td> <td> <span class="t-mark-rev t-since-cxx23">(since C++23)</span> </td> </tr> <tr class="t-dcl t-since-cxx23"> <td> <pre data-language="cpp">template&lt; class F &gt;
constexpr auto transform( F&amp;&amp; f ) const&amp;&amp;;</pre>
</td> <td> (4) </td> <td> <span class="t-mark-rev t-since-cxx23">(since C++23)</span> </td> </tr>  </table> <p>If <code>*this</code> contains an expected value, invokes <code>f</code> and returns a <code>std::expected</code> object that contains its result; otherwise, returns a <code>std::expected</code> object that contains a copy of <a href="error.html" title="cpp/utility/expected/error"><code>error()</code></a>.</p>
<p>If <code>T</code> is not (possibly cv-qualified) <span class="kw4">void</span>, the contained value (obtained from <a href="operator*.html" title="cpp/utility/expected/operator*"><code>operator*</code></a>) is passed as an argument to <code>f</code>; otherwise <code>f</code> takes no argument.</p>
<span class="t-li">1,2)</span> Given type <code>U</code> as <ul>
<li> <a href="http://en.cppreference.com/w/cpp/types/remove_cv"><span class="kw600">std::<span class="me2">remove_cv_t</span></span></a><span class="sy1">&lt;</span><a href="http://en.cppreference.com/w/cpp/types/result_of"><span class="kw3219">std::<span class="me2">invoke_result_t</span></span></a><span class="sy1">&lt;</span>F<span class="sy1">&gt;&gt;</span> if <code>T</code> is (possibly cv-qualified) <span class="kw4">void</span>, or </li>
<li> <a href="http://en.cppreference.com/w/cpp/types/remove_cv"><span class="kw600">std::<span class="me2">remove_cv_t</span></span></a><span class="sy1">&lt;</span><a href="http://en.cppreference.com/w/cpp/types/result_of"><span class="kw3219">std::<span class="me2">invoke_result_t</span></span></a><span class="sy1">&lt;</span>F, decltype<span class="br0">(</span><span class="sy2">**</span>this<span class="br0">)</span><span class="sy1">&gt;&gt;</span> otherwise,</li>
</ul>  and expression <code>/* invoke-expr */</code> as <ul>
<li> <code><a href="http://en.cppreference.com/w/cpp/utility/functional/invoke"><span class="kw1063">std::<span class="me2">invoke</span></span></a><span class="br0">(</span><a href="http://en.cppreference.com/w/cpp/utility/forward"><span class="kw1120">std::<span class="me2">forward</span></span></a><span class="sy1">&lt;</span>F<span class="sy1">&gt;</span><span class="br0">(</span>f<span class="br0">)</span><span class="br0">)</span></code> if <code>T</code> is (possibly cv-qualified) <span class="kw4">void</span>, or </li>
<li> <code><a href="http://en.cppreference.com/w/cpp/utility/functional/invoke"><span class="kw1063">std::<span class="me2">invoke</span></span></a><span class="br0">(</span><a href="http://en.cppreference.com/w/cpp/utility/forward"><span class="kw1120">std::<span class="me2">forward</span></span></a><span class="sy1">&lt;</span>F<span class="sy1">&gt;</span><span class="br0">(</span>f<span class="br0">)</span>, <span class="sy2">**</span>this<span class="br0">)</span></code> otherwise.</li>
</ul> <div class="t-li1">
 If <code>U</code> is not a valid value type for <code>std::expected</code>, or <code>U u(/* invoke-expr */);</code> is ill-formed when <code><a href="http://en.cppreference.com/w/cpp/types/is_void"><span class="kw458">std::<span class="me2">is_void_v</span></span></a><span class="sy1">&lt;</span>U<span class="sy1">&gt;</span></code> is <code>false</code>, the program is ill-formed.</div> <div class="t-li1">
 The effect is equivalent to <div class="cpp source-cpp"><pre data-language="cpp">if (has_value())
{
    if constexpr (std::is_void_v&lt;U&gt;)
    {
        /* invoke-expr */;
        return std::expected&lt;U, E&gt;();
    }
    else
        // the returned std::expected object contains an expected value,
        // which is direct-non-list-initialized with /* invoke-expr */
        return /* an std::expected&lt;U, E&gt; object */;
}
else
    return std::expected&lt;U, E&gt;(std::unexpect, error());</pre></div>
</div> <div class="t-li1">
 These overloads participate in overload resolution only if <code><a href="http://en.cppreference.com/w/cpp/types/is_constructible"><span class="kw528">std::<span class="me2">is_constructible_v</span></span></a><span class="sy1">&lt;</span>E, decltype<span class="br0">(</span>error<span class="br0">(</span><span class="br0">)</span><span class="br0">)</span><span class="sy1">&gt;</span></code> is <code>true</code>.</div> <span class="t-li">3,4)</span> Given type <code>U</code> as <ul>
<li> <a href="http://en.cppreference.com/w/cpp/types/remove_cv"><span class="kw600">std::<span class="me2">remove_cv_t</span></span></a><span class="sy1">&lt;</span><a href="http://en.cppreference.com/w/cpp/types/result_of"><span class="kw3219">std::<span class="me2">invoke_result_t</span></span></a><span class="sy1">&lt;</span>F<span class="sy1">&gt;&gt;</span> if <code>T</code> is (possibly cv-qualified) <span class="kw4">void</span>, or </li>
<li> <a href="http://en.cppreference.com/w/cpp/types/remove_cv"><span class="kw600">std::<span class="me2">remove_cv_t</span></span></a><span class="sy1">&lt;</span><a href="http://en.cppreference.com/w/cpp/types/result_of"><span class="kw3219">std::<span class="me2">invoke_result_t</span></span></a><span class="sy1">&lt;</span>F, decltype<span class="br0">(</span>std<span class="sy4">::</span><span class="me2">move</span><span class="br0">(</span><span class="sy2">**</span>this<span class="br0">)</span><span class="br0">)</span><span class="sy1">&gt;&gt;</span> otherwise,</li>
</ul>  and expression <code>/* invoke-expr */</code> as <ul>
<li> <code><a href="http://en.cppreference.com/w/cpp/utility/functional/invoke"><span class="kw1063">std::<span class="me2">invoke</span></span></a><span class="br0">(</span><a href="http://en.cppreference.com/w/cpp/utility/forward"><span class="kw1120">std::<span class="me2">forward</span></span></a><span class="sy1">&lt;</span>F<span class="sy1">&gt;</span><span class="br0">(</span>f<span class="br0">)</span><span class="br0">)</span></code> if <code>T</code> is (possibly cv-qualified) <span class="kw4">void</span>, or </li>
<li> <code><a href="http://en.cppreference.com/w/cpp/utility/functional/invoke"><span class="kw1063">std::<span class="me2">invoke</span></span></a><span class="br0">(</span><a href="http://en.cppreference.com/w/cpp/utility/forward"><span class="kw1120">std::<span class="me2">forward</span></span></a><span class="sy1">&lt;</span>F<span class="sy1">&gt;</span><span class="br0">(</span>f<span class="br0">)</span>, std<span class="sy4">::</span><span class="me2">move</span><span class="br0">(</span><span class="sy2">**</span>this<span class="br0">)</span><span class="br0">)</span></code> otherwise.</li>
</ul> <div class="t-li1">
 If <code>U</code> is not a valid value type for <code>std::expected</code>, or <code>U u(/* invoke-expr */);</code> is ill-formed when <code><a href="http://en.cppreference.com/w/cpp/types/is_void"><span class="kw458">std::<span class="me2">is_void_v</span></span></a><span class="sy1">&lt;</span>U<span class="sy1">&gt;</span></code> is <code>false</code>, the program is ill-formed.</div> <div class="t-li1">
 The effect is equivalent to <div class="cpp source-cpp"><pre data-language="cpp">if (has_value())
{
    if constexpr (std::is_void_v&lt;U&gt;)
    {
        /* invoke-expr */;
        return std::expected&lt;U, E&gt;();
    }
    else
        // the returned std::expected object contains an expected value,
        // which is direct-non-list-initialized with /* invoke-expr */
        return /* an std::expected&lt;U, E&gt; object */;
}
else
    return std::expected&lt;U, E&gt;(std::unexpect, std::move(error()));</pre></div> These overloads participate in overload resolution only if <code><a href="http://en.cppreference.com/w/cpp/types/is_constructible"><span class="kw528">std::<span class="me2">is_constructible_v</span></span></a><span class="sy1">&lt;</span>E, decltype<span class="br0">(</span>std<span class="sy4">::</span><span class="me2">move</span><span class="br0">(</span>error<span class="br0">(</span><span class="br0">)</span><span class="br0">)</span><span class="br0">)</span><span class="sy1">&gt;</span></code> is <code>true</code>.</div>  <h3 id="Parameters"> Parameters</h3> <table class="t-par-begin"> <tr class="t-par"> <td> f </td> <td> - </td> <td> a suitable function or <a href="../../named_req/callable.html" title="cpp/named req/Callable">Callable</a> object whose call signature returns a non-reference type </td>
</tr>
</table> <h3 id="Return_value"> Return value</h3> <p>A <code>std::expected</code> object containing either the result of <code>f</code> or an error value, as described above.</p>
<h3 id="Example"> Example</h3>  <h3 id="Defect_reports"> Defect reports</h3> <p>The following behavior-changing defect reports were applied retroactively to previously published C++ standards.</p>
<table class="dsctable"> <tr> <th>DR </th> <th>Applied to </th> <th>Behavior as published </th> <th>Correct behavior </th>
</tr> <tr> <td>
<a rel="nofollow" class="external text" href="https://cplusplus.github.io/LWG/issue3938">LWG 3938</a> </td> <td>C++23 </td> <td>
<code>transform</code> was ill-formed if <code>T</code> is not (possibly<br>cv-qualified) <span class="kw4">void</span> and <code>E</code> is not copyable </td> <td>made well-formed </td>
</tr>
</table> <h3 id="See_also"> See also</h3> <table class="t-dsc-begin"> <tr class="t-dsc"> <td> <div><a href="transform_error.html" title="cpp/utility/expected/transform error"> <span class="t-lines"><span>transform_error</span></span></a></div> </td> <td> returns the <code>expected</code> itself if it contains an expected value; otherwise, returns an <code>expected</code> containing the transformed unexpected value <br> <span class="t-mark">(public member function)</span>  </td>
</tr> </table>           <div class="_attribution">
  <p class="_attribution-p">
    &copy; cppreference.com<br>Licensed under the Creative Commons Attribution-ShareAlike Unported License v3.0.<br>
    <a href="https://en.cppreference.com/w/cpp/utility/expected/transform" class="_attribution-link">https://en.cppreference.com/w/cpp/utility/expected/transform</a>
  </p>
</div>
